File: src/Models/PlaceholderDefinition.cs
using YamlDotNet.Serialization;

namespace PromptSpec.Models
{

    /// <summary>
    /// Defines the validation rules for a placeholder in a prompt template.
    /// </summary>
    public class PlaceholderDefinition
    {
        /// <summary>
        /// The expected data type for this placeholder (string, number, boolean).
        /// </summary>
        [YamlMember(Alias = "type")]
        public string Type { get; set; } = "string";

        /// <summary>
        /// Indicates whether this placeholder is required and must be provided.
        /// </summary>
        [YamlMember(Alias = "required")]
        public bool Required { get; set; } = false;
    }
}

File: python/promptspec/models/placeholder_definition.py
"""
Placeholder definition for validation rules.
"""
from typing import Literal
from pydantic import BaseModel, Field

PlaceholderType = Literal["string", "number", "boolean"]


class PlaceholderDefinition(BaseModel):
    """Defines validation rules for a placeholder in a prompt template."""
    
    type: PlaceholderType = Field(
        "string", 
        description="Expected data type for this placeholder"
    )
    required: bool = Field(
        False, 
        description="Whether this placeholder is required"
    )